receiver/elasticapmreceiver/internal/mappers/intakeV2ToOtlpTopLevelFields.go

```diff
@@ -0,0 +1,116 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mappers
+
+import (
+	"encoding/hex"
+	"strings"
+	"time"
+
+	"github.com/elastic/apm-data/model/modelpb"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type TopLevelFieldSetter interface {
+	SetTraceID(v pcommon.TraceID)
+	SetSpanID(v pcommon.SpanID)
+}
+
+// Shared across LogRecord and Span
+func SetTopLevelFieldsCommon(event *modelpb.APMEvent, timestamp time.Time, t TopLevelFieldSetter, logger *zap.Logger) {
```

Comment:
Is the `timestamp` parameter needed here?

Reply:
No, you are right - removed 👍